internal/tfdiags/consolidate_warnings_test.go

```diff
@@ -96,7 +96,7 @@ func TestConsolidateWarnings(t *testing.T) {
 
 	// We're using ForRPC here to force the diagnostics to be of a consistent
 	// type that we can easily assert against below.
-	got := diags.ConsolidateWarnings(2).ForRPC()
+	got := diags.Consolidate(2, Warning).ForRPC()
```

Comment:
nit: I would prefer constants for `1` and `2` when calling `Consolidate`. It isn't clear to me why exactly this number should be passed.

Reply:
Other than this test, it's always 1.  The function was originally built to support consolidation at different count thresholds.  I see this as a legacy option that I didn't want to touch in this PR, even though the arg should probably be removed.